Issue#461

stepanzharychevbroadcom opened the pull request:
Implementation for use cases listed in https://github.com/zowe/vscode-extension-for-zowe/issues/461#issuecomment-578194128

Colin-Stone commented on src/utils/uss.ts:
```diff
@@ -0,0 +1,29 @@
+export function injectAdditionalDataToTooltip(node: ZoweUSSNode, tooltip: string) {
+    if (node.downloaded && node.downloadedTime) {
+        // TODO: Add time formatter to localization so we will use not just US variant
+        return `${tooltip} (Last Pulled: ${moment(node.downloadedTime).format("HH:mm MM/DD/YY")})`;
```
Really useful tooltip. Just a question. "Last pulled" or "Last downloaded"

stepanzharychevbroadcom replied:
I guess they have equal meaning. First is just shorter.

Colin-Stone replied:
I think Downloaded would probably mean more to most users. It could always be simply "Downloaded" instead of "Last Pulled" ?.

stepanzharychevbroadcom replied:
Yes, sure we can.

stepanzharychevbroadcom commented:
> Really neat function.I did find something weird though. I downloaded a UTF-8 file. Which was converted into gibberish. Hit the Toggle Binary button and everything was perfect, icon was right etc. When I clicked the file again it redownloaded it in gibberish. i.e. didn't maintain the binary decision. Same thing if I use the cloud download button.
> 
> Got a `Cannot read property 'setEtag' of undefined` popup when saving a file I had switched to binary

@Colin-Stone I tried it with image and it works. In the case if you changed your filter (e.g. from one folder to another) this decision won't be kept, it's not yet implemented.
For "Downloaded text" I added fix.

stepanzharychevbroadcom commented:
@Colin-Stone @Alexandru-Dumitru Could you take a look again? I would really like to have it merged before more changes in master will happen.

Colin-Stone commented:
Is it me or have the icon defintion changed?. They are paler, less well defined than those around them and resemble the reversed icons technique I use for indicating migrated datasets currently. i.e the normal file icon for a dark theme used in the light theme and vice-versa. These are the file (text and binary ones)
![image](https://user-images.githubusercontent.com/30794003/73455791-a610cc80-4368-11ea-93fb-ebb2983bc650.png) 
![image](https://user-images.githubusercontent.com/30794003/73455841-b628ac00-4368-11ea-809e-dc5810df9fab.png)
![image](https://user-images.githubusercontent.com/30794003/73455875-c6408b80-4368-11ea-8a11-7abf1f489308.png)

stepanzharychevbroadcom commented:
> Is it me or have the icon defintion changed?. They are paler, less well defined than those around them and resemble the reversed icons technique I use for indicating migrated datasets currently. i.e the normal file icon for a dark theme used in the light theme and vice-versa. These are the file (text and binary ones)
> ![image](https://user-images.githubusercontent.com/30794003/73455791-a610cc80-4368-11ea-93fb-ebb2983bc650.png)
> ![image](https://user-images.githubusercontent.com/30794003/73455841-b628ac00-4368-11ea-809e-dc5810df9fab.png)
> ![image](https://user-images.githubusercontent.com/30794003/73455875-c6408b80-4368-11ea-8a11-7abf1f489308.png)

@Colin-Stone No, that's how it should be, once the file is downloaded we decrease the opacity of icon so it will be additionally understandable for the user that the file is different. It's 30% opacity right now. We came to this solution together with UI designer.

Alexandru-Dumitru commented on src/ZoweUSSNode.ts:
```diff
@@ -216,22 +232,69 @@ export class ZoweUSSNode extends vscode.TreeItem implements IZoweTreeNode {
+     */
+    public get isDirtyInEditor(): boolean {
+        const openedTextDocuments = vscode.workspace.textDocuments;
+        const currentFilePath = getUSSDocumentFilePath(this);
```
Why not `extension.getUSSDocumentFilePath(this);` and avoid duplicate imports?

Alexandru-Dumitru commented on src/ZoweUSSNode.ts:
```diff
@@ -216,22 +232,69 @@ export class ZoweUSSNode extends vscode.TreeItem implements IZoweTreeNode {
+
+    public get openedDocumentInstance(): vscode.TextDocument {
+        const openedTextDocuments = vscode.workspace.textDocuments;
+        const currentFilePath = getUSSDocumentFilePath(this);
```
same here

Alexandru-Dumitru commented on src/ZoweUSSNode.ts:
```diff
@@ -12,11 +12,21 @@
 import { Session } from "@brightside/imperative";
 import * as vscode from "vscode";
+// tslint:disable-next-line: no-duplicate-imports
+import { TreeItemCollapsibleState } from "vscode";
```
This line is redundant, because `TreeItemCollapsibleState` is never used directly.

Alexandru-Dumitru commented on src/ZoweUSSNode.ts:
```diff
@@ -12,11 +12,21 @@
 import * as utils from "./utils";
+import { getIconByNode } from "./generators/icons/index";
+// tslint:disable-next-line: no-implicit-dependencies
+import * as moment from "moment";
```
Shouldn't we add this as dependency instead of `momentjs` ?

Alexandru-Dumitru commented on src/ZoweUSSNode.ts:
```diff
@@ -250,10 +308,37 @@ export class ZoweUSSNode extends vscode.TreeItem implements IZoweTreeNode {
     /**
-     * helper method to change the node names in one go
-     * @param oldReference string
-     * @param revision string
```
Not sure why part of documentation jumped inside the body of the function 😄

stepanzharychevbroadcom replied:
@Alexandru-Dumitru Fixed.
